flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/hybrid/HybridSourceReader.java

```diff
@@ -60,11 +62,20 @@ public HybridSourceReader(
 
     @Override
     public void start() {
-        setCurrentReader(0);
+        // underlying reader starts on demand with split assignment
     }
 
     @Override
     public InputStatus pollNext(ReaderOutput output) throws Exception {
```

Comment:
Do we want to support a bounded hybrid? Currently, there is no `END_OF_INPUT` return value...